init_windows.go

```diff
@@ -0,0 +1,36 @@
+//go:build !cmd_go_bootstrap
+
+package openssl
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+type dlopenError struct {
+	file string
+	err  error
+}
+
+func (e *dlopenError) Error() string {
+	return "openssl: can't load " + e.file + ": " + e.err.Error()
+}
+
+func (e *dlopenError) Unwrap() error {
+	return e.err
+}
+
+func dlopen(file string) (handle unsafe.Pointer, err error) {
+	// As Windows generally does not ship with a system OpenSSL library, let
+	// alone a FIPS 140 certified one, use the default library search order so
+	// that we preferentially load the DLL bundled with the application.
+	h, err := syscall.LoadLibrary(file)
+	if err != nil {
+		return nil, &dlopenError{file: file, err: err}
```

Comment:
Why returning a typed error instead of just `errors.New("openssl: can't load " + e.file + ": " + err.Error())`

Reply:
So the caller (of `openssl.Init()`) is able to interrogate the wrapped `syscall.Errno` error returned from `LoadLibrary` for the cause of the failure, e.g. to differentiate `errors.Is(err, syscall.ERROR_ACCESS_DENIED)` from `errors.Is(err, syscall.ERROR_MOD_NOT_FOUND)`.

In contrast, POSIX `dlopen()` only surfaces errors through the `dlerror()` API, which returns a human readable string rather than an enumerated error. The caller would not be able to do much programmatically with that information aside from surfacing it up to the user verbatim, so nothing of value is lost by returning a plain `errors.New()` string there.

Reply:
Thanks, makes sense.